src/tests.rs

```diff
@@ -131,3 +133,77 @@ fn probe_constant() {
     ac.assert_min(1, 33, ac.probe_constant("{ let x = 1 + 2 + 3; x * x }"));
     ac.assert_min(1, 39, ac.probe_constant(r#""test".len()"#));
 }
+
+#[test]
+fn dir_does_not_contain_target() {
+    let _set_cargo_target_dir = SetVar::new("CARGO_TARGET_DIR", None);
```

Comment:
The environment is a process-wide resource, so this will race when the tests are run in parallel. I ran tests in a loop locally and they do indeed fail occasionally.

You could add a static `Mutex` and put a locked `MutexGuard` in `SetVar`, effectively serializing them. Or you could make `CARGO_TARGET_DIR` a parameter of the function too.

Reply:
Thanks for catching! I went with making the value of `$CARGO_TARGET_DIR` a parameter as the result is overall simpler.